lib/runtime/dart/_interceptors.js

```diff
@@ -391,7 +391,7 @@ dart_library.library('dart/_interceptors', null, /* Imports */[
         return new (_internal.IterableMixinWorkaround$(E))().asMapList(this);
       }
     }
-    dart.setBaseClass(JSArray, dart.global.Array);
+    dart.setBaseClass(JSArray, (typeof global == 'undefined' ? dart.global : global).Array);
```

Comment:
Actually, it all works without it, not sure what led to this TBH. I was having difficulties aliasing global to dart.global (dart.global.Array would always return undefined). Might have been fixed by the `let window = window;` fix.